### PR TITLE
Move classes from spark to async-query-core and async-query

### PR DESCRIPTION
**Pull request by ykmr1224**
### Description
- Move classes from spark to async-query-core and async-query to separate async-query-core as library.
- There are still some dependency from async-query-core to datasources/legacy/spark, and those will be addressed in following PRs.
- Since many classes were covered by integration tests, classes in async-query-core module are currently not covered by tests in async-query-core. Unit tests and integration tests for async-query-core will be added in following PRs. (uncovered one are excluded from coverage check right now)
 
### Issues Resolved
n/a
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: async-query-core/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 1]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (seankao-az) [inline async-query-core/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: what is this for? and nit: add newline in the end

File: async-query/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 2]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (seankao-az) [inline async-query/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: same

File: async-query-core/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker (hunk 1 above)
Comment (ykmr1224) [reply async-query-core/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: This is a magic spell to enable mocking on final class with Mockito. (Mockito disable it by default, and can be enabled with this configuration file. And we are doing the same for other modules.)

File: async-query/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker (hunk 2 above)
Comment (ykmr1224) [reply async-query/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: same as above.

**Review by seankao-az — APPROVED** [review]

**Review by vamsi-amazon — APPROVED** [review]

**Pull request merged**